### PR TITLE
tests: rpl: convert legacy test in ztest

### PR DESCRIPTION
**Pull request by punitvara**
This patch makes use of ztest apis to make output unified with
other test cases

Signed-off-by: Punit Vara <punit.vara@intel.com>

### CODE REVIEW

**Review by jukkar — APPROVED** [review]

File: tests/net/rpl/src/main.c [hunk 1]
```diff
@@ -299,22 +280,19 @@ static bool test_rpl_mcast_addr(void)
...
+		zassert_true(0, "RPL multicast address check failed.");
 	}
 
 	net_rpl_create_mcast_address(&addr);
 
 	ret = net_rpl_is_ipv6_addr_mcast(&addr);
 	if (!ret) {
-		TC_ERROR("Generated RPL multicast address check failed.\n");
-		return false;
+		zassert_true(0, "Generated RPL multicast "
```
Comment (andrewboie) [inline tests/net/rpl/src/main.c]: don't split strings even if they go past 80 characters, it is much harder to grep for them.
checkpatch makes an exception for strings like this.

File: tests/net/rpl/src/main.c [hunk 2]
```diff
@@ -510,7 +467,7 @@ static bool populate_nbr_cache(void)
 
 	if (data_failure) {
 		data_failure = false;
-		return false;
+		zassert_true(0, NULL);
```
Comment (andrewboie) [inline tests/net/rpl/src/main.c]: can't you just zassert on data_failure directly?

File: tests/net/rpl/src/main.c [hunk 3]
```diff
@@ -501,7 +458,7 @@ static bool populate_nbr_cache(void)
 	data_failure = false;
 
 	if (!net_test_send_ns()) {
-		return false;
+		zassert_true(0, NULL);
```
Comment (andrewboie) [inline tests/net/rpl/src/main.c]: zassert on net_test_send_ns directly

File: tests/net/rpl/src/main.c [hunk 4]
```diff
@@ -520,18 +477,12 @@ static bool populate_nbr_cache(void)
...
 			       NET_IPV6_NBR_STATE_REACHABLE);
-	if (!nbr) {
-		TC_ERROR("Cannot add peer to neighbor cache\n");
-		return false;
-	}
+	zassert_not_null(nbr, "Cannot add peer to neighbor cache");
 
 	if (!net_test_nbr_lookup_ok()) {
-		return false;
+		zassert_true(0, NULL);
```
Comment (andrewboie) [inline tests/net/rpl/src/main.c]: as above

File: tests/net/rpl/src/main.c [hunk 5]
```diff
@@ -647,77 +598,41 @@ static bool test_dio_receive_dest(void)
...
-	}
+	zassert_false(ret, "cannot send DIO");
 
 	k_sem_take(&wait_data, WAIT_TIME);
 
 	if (data_failure) {
 		data_failure = false;
-		TC_ERROR("%d: Unexpected ICMPv6 code received\n", __LINE__);
-		return false;
+		zassert_true(0, "Unexpected ICMPv6 code received");
```
Comment (andrewboie) [inline tests/net/rpl/src/main.c]: as above

**Review by tyrantlizard77 — CHANGES_REQUESTED** [review]

File: tests/net/rpl/src/main.c (hunk 1 above)
Comment (punitvara) [reply tests/net/rpl/src/main.c]: fixed

File: tests/net/rpl/src/main.c (hunk 3 above)
Comment (punitvara) [reply tests/net/rpl/src/main.c]: fixed

File: tests/net/rpl/src/main.c (hunk 2 above)
Comment (punitvara) [reply tests/net/rpl/src/main.c]: fixed

File: tests/net/rpl/src/main.c (hunk 4 above)
Comment (punitvara) [reply tests/net/rpl/src/main.c]: fixed

File: tests/net/rpl/src/main.c (hunk 5 above)
Comment (punitvara) [reply tests/net/rpl/src/main.c]: fixed

Comment (punitvara) [reply tests/net/rpl/src/main.c]: @andrewboie Can you please look at this PR once ?

**Review by tyrantlizard77 — APPROVED** [review]

**Pull request merged**